agent/taskresource/firelens/firelens_unix.go

```diff
@@ -94,33 +102,38 @@ type FirelensResource struct {
 }
 
 // NewFirelensResource returns a new FirelensResource.
-func NewFirelensResource(cluster, taskARN, taskDefinition, ec2InstanceID, dataDir, firelensConfigType, region, networkMode string,
-	firelensOptions map[string]string, containerToLogOptions map[string]map[string]string, credentialsManager credentials.Manager,
-	executionCredentialsID string) (*FirelensResource, error) {
-	firelensResource := &FirelensResource{
-		cluster:                cluster,
-		taskARN:                taskARN,
-		taskDefinition:         taskDefinition,
-		ec2InstanceID:          ec2InstanceID,
-		firelensConfigType:     firelensConfigType,
-		region:                 region,
-		networkMode:            networkMode,
-		containerToLogOptions:  containerToLogOptions,
-		ioutil:                 ioutilwrapper.NewIOUtil(),
-		s3ClientCreator:        factory.NewS3ClientCreator(),
-		executionCredentialsID: executionCredentialsID,
-		credentialsManager:     credentialsManager,
-	}
+func NewFirelensResource(cluster, taskARN, taskDefinition, ec2InstanceID, dataDir, region, networkMode string,
+	firelensConfig *apicontainer.FirelensConfig, containerToLogOptions map[string]map[string]string,
+	credentialsManager credentials.Manager, executionCredentialsID string) (*FirelensResource, error) {
 
 	fields := strings.Split(taskARN, "/")
 	taskID := fields[len(fields)-1]
-	firelensResource.resourceDir = filepath.Join(filepath.Join(dataDir, "firelens"), taskID)
 
-	err := firelensResource.parseOptions(firelensOptions)
-	if err != nil {
-		return nil, errors.Wrap(err, "error parsing firelens options")
-	}
+	var firelensResource *FirelensResource
+	if firelensConfig.Version != "v2" {
+		firelensResource = &FirelensResource{
```

Comment:
I think if we don't initialize and assign `resourceDir` and `firelensConfig` fields for v2, it may throw a nil pointer exception. also looks like in unit tests, it is explicitly initialized. 

you could do something like:
```

firelensResource = &FirelensResource{}
if if firelensConfig.Version != "v2" { 
   // add v1 fields 
} else {
   // add v2 fields
}
```

Reply:
oh resourceDir and firelensConfig are being initialized on line 138 and 140 for firelens v2.